parser.py

```diff
@@ -459,13 +506,14 @@ def uop(self, *args):
             # <id> <anylist> , <uop>
             # <id>() <anylist>, <uop>
             # <id>(<explist>) <anylist>, <uop>
-            params += other_op.params
-            wires += other_op.wires
-            op_name = "".join([op_name, other_op.name])
+            op = Op(op_name, params, wires)
+            return TensorOp(op, other_op)
 
         if extra_args[0].name == "idlist":
             # TODO: this case should not occur, according
             # to the original spec. Determine what is causing it
+            # @co9olguy: this is because self.gatedecl calls
+            # this method, and self.gatedecl *can* have idlist.
```

Comment:
@co9olguy this was a time-saving 'hack' on my side, just to avoid code duplication. Perhaps the crux of this method should be factored out as an external function/another private method, which can be called by both `self.uop()` and `self.gatedecl()`.